File: db4on/tutorial/db4o-tutorial-chapters/src/com/db4o/f1/chapter6/EvenHistoryEvaluation.cs
using com.db4o.f1.chapter3;
using com.db4o.query;
	
namespace com.db4o.f1.chapter6
{	
	public class EvenHistoryEvaluation : Evaluation
	{
		public void Evaluate(Candidate candidate)
		{
			Car car=(Car)candidate.GetObject();
			candidate.Include(car.History.Count % 2 == 0);
		}
	}
}

File: db4on/tutorial/vb/db4o-tutorial-chapters/src/com/db4o/f1/chapter6/EvenHistoryEvaluation.vb
Imports com.db4o.query
Namespace com.db4o.f1.chapter6
	Public Class EvenHistoryEvaluation
        Implements Evaluation
        Public Sub Evaluate(ByVal candidate As Candidate) Implements Evaluation.Evaluate
            Dim car As chapter3.Car = DirectCast(candidate.GetObject(), chapter3.Car)
            candidate.Include(car.History.Count Mod 2 = 0)
        End Sub

	End Class
End Namespace
